pkg/controller/hostpathprovisioner/daemonset.go

```diff
@@ -191,3 +195,13 @@ func createDaemonSetObject(cr *hostpathprovisionerv1.HostPathProvisioner, provis
 		},
 	}
 }
+
+// updateDaemonSetObject returns a DaemonSet with values set from CR.
+func updateDaemonSetObject(cr *hostpathprovisionerv1.HostPathProvisioner, orig runtime.Object) runtime.Object {
+	newDaemonSet := orig.(*appsv1.DaemonSet)
+	newDaemonSet.Spec.Template.Spec.NodeSelector = cr.Spec.Workloads.NodeSelector
+	newDaemonSet.Spec.Template.Spec.Tolerations = cr.Spec.Workloads.Tolerations
+	newDaemonSet.Spec.Template.Spec.Affinity = cr.Spec.Workloads.Affinity
+
+	return newDaemonSet.DeepCopyObject()
```

Comment:
Are you sure you want to update `orig` and return a copy of it?  I think you would either want to only mutate the copy or mutate orig and return nothing.